### PR TITLE
future possibilities and consequences

### PR DESCRIPTION
**Pull request by wacban**

### CODE REVIEW

File: neps/nep-0509.md [hunk 1]
```diff
@@ -473,34 +473,34 @@ This makes chunk validator selection algorithm correctness critical for the secu
...
 ### Neutral
 
-- n1
+- The current approach to resharding will need to be revised to support generating state witness. 
+- The security assumptions will change. The responsibility will be moved from block producers to chunk validators and the security will become probabilistic. 
 
 ### Negative
 
-- n1
+- The hardware requirements will increase. In particular the network bandwidth and memory requirements will be higher.
```
Comment (walnut-the-cat) [inline neps/nep-0509.md:501]: hmm wonder if we should mention potential decrease of hardware requirement (disk requirement?) as well in positive section

File: neps/nep-0509.md [hunk 2]
```diff
@@ -473,34 +473,34 @@ This makes chunk validator selection algorithm correctness critical for the secu
...
-- n1
+- The current approach to resharding will need to be revised to support generating state witness. 
+- The security assumptions will change. The responsibility will be moved from block producers to chunk validators and the security will become probabilistic. 
 
 ### Negative
 
-- n1
+- The hardware requirements will increase. In particular the network bandwidth and memory requirements will be higher. 
+- Additional limits will be put on the size of transactions, receipts and, more generally, cross shard communication.
+- The dependency on cloud state sync will increase the centralization of the blockchain. This will be resolved separately by the decentralized state sync.
```
Comment (walnut-the-cat) [inline neps/nep-0509.md:503]: should we specify what it starts having a dependency on state sync?

Comment (Longarithm) [reply neps/nep-0509.md:503]: Well, it's mentioned above on explanation of validator roles.

File: neps/nep-0509.md (hunk 1 above)
Comment (Longarithm) [reply neps/nep-0509.md:501]: +1 - more specifically, small-stake nodes won't have to store state and apply chunks for all shards as they currently do.

**Comment by wacban** [comment]
@walnut-the-cat and @Longarithm I added the suggestion and I'm going to merge it now. Feel free to add any more comments and suggestions and I'll just do it in a follow up PR.

**Pull request merged**